src/glue/jobs/s3_to_json.py

```diff
@@ -81,51 +81,78 @@ def transform_object_to_array_of_objects(
         # ]
     """
     array_of_obj = []
-    value_error = "Failed to cast %s to %s. Setting value to None."
     for k, v in json_obj_to_replace.items():
         try:
             key_value = key_type(k)
         except ValueError as error:
-            logger.error(
-                    value_error, k, key_type,
-                    extra={
-                        **logger_context,
-                        "error.message": repr(error),
-                        "error.type": "ValueError",
-                        "event.kind": "alert",
-                        "event.category": ["configuration"],
-                        "event.type": ["deletion"],
-                        "event.outcome": "failure"
-                    }
-            )
             key_value = None
+            _log_error_transform_object_to_array_of_objects(
+                    value=k,
+                    value_type=key_type,
+                    error=error,
+                    logger_context=logger_context
+            )
         try:
             value_value = value_type(v)
         except ValueError as error:
-            logger.error(
-                    value_error, v, value_type,
-                    extra={
-                        **logger_context,
-                        "error.message": repr(error),
-                        "error.type": "ValueError",
-                        "event.kind": "alert",
-                        "event.category": ["configuration"],
-                        "event.type": ["deletion"],
-                        "event.outcome": "failure"
-                    }
-            )
             value_value = None
+            _log_error_transform_object_to_array_of_objects(
+                    value=v,
+                    value_type=value_type,
+                    error=error,
+                    logger_context=logger_context
+            )
         obj = {
             key_name: key_value,
             value_name: value_value
         }
         array_of_obj.append(obj)
     return array_of_obj
 
+def _log_error_transform_object_to_array_of_objects(
```

Comment:
+1 to this helper function, by refactoring, the code also becomes "easier" to write unit tests for.